### PR TITLE
test: fallback google.com check for Windows outbound test

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Adds the same "check google.com in case" functionality for Windows outbound test to match the Linux outbound test.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@PatrickLang @craiglpeters FYI

File: test/e2e/kubernetes/pod/pod.go [hunk 1]
```diff
@@ -692,12 +694,12 @@ func (p *Pod) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool,
...
 				}
 				// if we can curl bing.com we have outbound internet access
-				out, err := p.Exec("--", "curl", "bing.com")
+				out, err := p.Exec("--", "curl", externalURLs[0])
 				if err == nil {
 					readyCh <- true
 				} else {
 					// in case bing.com is down let's hope google.com is also not down
-					_, err := p.Exec("--", "curl", "google.com")
+					_, err := p.Exec("--", "curl", externalURLs[2])
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: should this be `externalURLs[1]`?

Comment (jackfrancis) [reply test/e2e/kubernetes/pod/pod.go]: lol

File: test/e2e/kubernetes/pod/pod.go [hunk 2]
```diff
@@ -818,8 +820,20 @@ func (p *Pod) CheckWindowsOutboundConnection(url string, sleep, duration time.Du
 						readyCh <- false
 					}
 				} else {
-					log.Printf("Error:%s\n", err)
-					log.Printf("Out:%s\n", out)
+					// in case bing.com is down let's hope google.com is also not down
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: I would remove this comment.

or make it something like "try with a second URL in the case the first one is down"

File: test/e2e/kubernetes/pod/pod.go [hunk 3]
```diff
@@ -692,12 +694,12 @@ func (p *Pod) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool,
 					installedCurl = true
 				}
 				// if we can curl bing.com we have outbound internet access
-				out, err := p.Exec("--", "curl", "bing.com")
+				out, err := p.Exec("--", "curl", externalURLs[0])
 				if err == nil {
 					readyCh <- true
 				} else {
 					// in case bing.com is down let's hope google.com is also not down
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: "let's hope google.com is not also down" or generalize comment

File: test/e2e/kubernetes/pod/pod.go [hunk 4]
```diff
@@ -1026,3 +1045,20 @@ func (c *Container) getMemoryRequests() string {
 func (c *Container) getMemoryLimits() string {
 	return c.Resources.Limits.Memory
 }
+
+// getExternalURL returns an external URL that is not in the passed-in array of strings
+func getExternalURL(urls []string) string {
+	m := map[string]bool{}
+	for _, url := range externalURLs {
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: the `externalURLs` var is only used here so it seems like it doesn't need to be a global var

Comment (jackfrancis) [reply test/e2e/kubernetes/pod/pod.go]: fair, the only reason I put it up there would be for easier maintenance, but we can make it private for now

File: test/e2e/kubernetes/pod/pod.go [hunk 5]
```diff
@@ -691,19 +691,17 @@ func (p *Pod) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool,
...
+				// if we can curl an external URL we have outbound internet access
+				for i, url := range getExternalURLs() {
+					out, err := p.Exec("--", "curl", url)
 					if err == nil {
 						readyCh <- true
 					} else {
-						// if both bing.com and google.com are down let's say we don't have outbound internet access
-						log.Printf("Error:%s\n", err)
-						log.Printf("Out:%s\n", out)
+						if i > 0 {
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: `if i == len(getExternalURLs) - 1` would scale when we add new URLs

File: test/e2e/kubernetes/pod/pod.go [hunk 6]
```diff
@@ -809,17 +807,19 @@ func (p *Pod) CheckWindowsOutboundConnection(url string, sleep, duration time.Du
...
+				// if we can curl an external URL we have outbound internet access
+				for i, url := range getExternalURLs() {
+					out, err := p.Exec("--", "powershell", "iwr", "-UseBasicParsing", "-TimeoutSec", "60", url)
 					matched := exp.MatchString(string(out))
-					if matched {
+					if err == nil && matched {
 						readyCh <- true
 					} else {
-						readyCh <- false
+						if i > 0 {
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: ditto

File: test/e2e/kubernetes/pod/pod.go [hunk 7]
```diff
@@ -691,19 +691,17 @@ func (p *Pod) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool,
...
+				for i, url := range getExternalURLs() {
+					out, err := p.Exec("--", "curl", url)
 					if err == nil {
 						readyCh <- true
 					} else {
-						// if both bing.com and google.com are down let's say we don't have outbound internet access
-						log.Printf("Error:%s\n", err)
-						log.Printf("Out:%s\n", out)
+						if i > 0 {
+							// if both are down let's say we don't have outbound internet access
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: if all* instead of both

File: test/e2e/kubernetes/pod/pod.go [hunk 8]
```diff
@@ -809,17 +807,19 @@ func (p *Pod) CheckWindowsOutboundConnection(url string, sleep, duration time.Du
...
+				for i, url := range getExternalURLs() {
+					out, err := p.Exec("--", "powershell", "iwr", "-UseBasicParsing", "-TimeoutSec", "60", url)
 					matched := exp.MatchString(string(out))
-					if matched {
+					if err == nil && matched {
 						readyCh <- true
 					} else {
-						readyCh <- false
+						if i > 0 {
+							// if both are down let's say we don't have outbound internet access
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: same

File: test/e2e/kubernetes/pod/pod.go [hunk 9]
```diff
@@ -691,19 +691,18 @@ func (p *Pod) CheckLinuxOutboundConnection(sleep, duration time.Duration) (bool,
...
+				urls := getExternalURLs()
+				for i, url := range urls {
+					out, err := p.Exec("--", "curl", url)
 					if err == nil {
 						readyCh <- true
 					} else {
-						// if both bing.com and google.com are down let's say we don't have outbound internet access
-						log.Printf("Error:%s\n", err)
-						log.Printf("Out:%s\n", out)
+						if i > (len(urls) - 1) {
```
Comment (CecileRobertMichon) [inline test/e2e/kubernetes/pod/pod.go]: this will never be true, i will never be > len of the array - 1.

it should be `==`

**Pull request merged**